### ISSUE TITLE
Resolve a11y issues on Address Confirmation Modal

### ISSUE THREAD

**Issue by georgehager** [CDCgov/prime-simplereport#4565]
## Background

We've added axe accessibility testing to our cypress e2e tests, and have some failures that need to be resolved. Currently, our Address Confirmation Modal triggers a11y errors across multiple pages, as shown by our e2e tests.

## Action Requested

- Remove the `rules` in the `checkA11y()` call that allow the axe cypress tests to ignore the Modal errors
- Fix the a11y issues on the Modal

## Acceptance Criteria

Running a Deque audit and the e2e tests result in no a11y errors related to the Address Confirmation Modal.

## Resources & Notes

- The Modal errors can be easily found in e2e specs 01, 02, 05, 08 with a test and comments similar to this:
```
    // failing a11y test
    // Also found in specs 01, 02, 05, 08
    // Test a11y on the confirm address modal
    cy.checkA11y(
        {
          exclude: [],
        },
        {
          rules: {
            'aria-dialog-name': { enabled: false },
            'landmark-one-main': { enabled: false },
            'page-has-heading-one': { enabled: false },
          },
        },
    );
```
- For the affected calls to `checka11y()` there may be multiple errors, not all of which are related to the Address Confirmation Modal. It's not necessary to fix those at this time.
